SecretSanta/src/SecretSanta.Api/Controllers/BaseApiController.cs

```diff
@@ -1,47 +1,48 @@
 ﻿using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
-using SecretSanta.Business;
-using SecretSanta.Data;
+using SecretSanta.Business.Services;
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace SecretSanta.Api.Controllers
 {
-    public abstract class BaseApiController<TEntity> : ControllerBase where TEntity : EntityBase
+    public abstract class BaseApiController<TDto, TInputDto> : ControllerBase
+        where TDto : class, TInputDto
+        where TInputDto : class
     {
-        protected IEntityService<TEntity> Service { get; }
+        protected IEntityService<TDto, TInputDto> Service { get; }
 
-        protected BaseApiController(IEntityService<TEntity> service)
+        protected BaseApiController(IEntityService<TDto, TInputDto> service)
         {
             Service = service ?? throw new ArgumentNullException(nameof(service));
         }
 
        [HttpGet]
-        public async Task<IEnumerable<TEntity>> Get() => await Service.FetchAllAsync();
+        public async Task<IEnumerable<TDto>> Get() => await Service.FetchAllAsync();
 
         [HttpGet("{id}")]
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
         [ProducesDefaultResponseType]
         public async Task<IActionResult> Get(int id)
         {
-            TEntity entity = await Service.FetchByIdAsync(id);
-            if (entity is null)
+            TDto tDto = await Service.FetchByIdAsync(id);
```

Comment:
In general we omit the "t" portion of the name.
```suggestion
            TDto dto = await Service.FetchByIdAsync(id);
```